lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2774,7 +2774,7 @@ void SPIRVToLLVM::transFunctionAttrs(SPIRVFunction *BF, Function *F) {
         AttrTy = cast<PointerType>(I->getType())->getPointerElementType();
         break;
       case Attribute::AttrKind::StructRet:
-        AttrTy = I->getType();
+        AttrTy = cast<PointerType>(I->getType())->getPointerElementType();
```

Comment:
Is it possible not to use `getPointerElementType` API here? As is it goes against the intentions of https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/1444

Reply:
I haven't looked at replacing the uses of `getPointerElementType` in this context yet--the next patch I have is replacing a dozen-ish uses with types scavenged form places like `sret`/`byval`/other parameters, which is where I discovered that the `sret` types were being improperly generated.

I think it would be better to handle the replacement of the `getPointerElementType` here at the same time as the one a few lines above rather than as a component of this patch in particular.

Reply:
Okay, LGTM me then